tests/conftest.py

```
@@ -3,218 +3,152 @@
-    
     env = os.environ.copy()
-    env['PYTHONUNBUFFERED'] = '1'  # Real-time output (no buffering)
+    env['PYTHONUNBUFFERED'] = '1'
+    
+    # สร้างไฟล์ชั่วคราวเพื่อเก็บ Log ของแอป
+    log_file = tempfile.NamedTemporaryFile(delete=False, mode='w+')
     
     try:
-        # Use 'python -m shiny run' instead of 'shiny run'
-        # This is more reliable across different Python installations
         process = subprocess.Popen(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)